pkg/controllers/flannelmigration/migration_controller.go

```diff
@@ -214,7 +231,7 @@ func (c *flannelMigrationController) processNewNode(node *v1.Node) {
 	}
 
 	log.Infof("Start processing new node %s.", node.Name)
-	err = c.ipamMigrator.SetupCalicoIPAMForNode(node)
+	err = c.ipamMigrator.SetupCalicoIPAMForNode(node, -1)
```

Comment:
Isn't this going to produce weird log messages which say we're processing index -1?

I feel like this will be confusing for users and future readers of this code. At a minimum we should comment what this value means and why it is here, but if possible I think it would be good to do something more clear (like moving that log with the index out of `SetupCalicoIPAMForNode` so the calling code can log out an appropriate message).

Reply:
Agree. It is a bit hacky to do that. I will make changes as you suggested.